packages/components/src/composer/composer.tsx

```diff
@@ -13,12 +15,29 @@ import type { GetProps } from '@tamagui/core'
 
 type BaseProps = GetProps<typeof YStack>
 
-const Composer = (props: BaseProps) => {
+const Composer = (
+  props: BaseProps & {
+    placeholderTextColor?: BaseProps['backgroundColor']
+    iconOptionsColor?: BaseProps['backgroundColor']
+    isBlurred?: boolean
+  }
+) => {
+  const { backgroundColor, isBlurred, ...rest } = props
+
+  const [isFocused, setIsFocused] = useState(false)
+
+  const applyVariantStyles:
+    | {
+        blurred: boolean
+      }
+    | undefined = isBlurred && !isFocused ? { blurred: true } : undefined
+
   return (
     <YStack
-      backgroundColor="$background"
-      shadowColor="rgba(9, 16, 28, 0.08)"
-      shadowOffset={{ width: 0, height: -4 }}
+      animation="fast"
+      backgroundColor={isFocused ? '$background' : backgroundColor}
+      shadowColor={!isBlurred || isFocused ? 'rgba(9, 16, 28, 0.08)' : 'none'}
+      shadowOffset={{ width: 4, height: !isBlurred || isFocused ? 4 : 0 }}
```

Comment:
I think this condition should be unified into one place.

Reply:
`isBlurred` is coming from props with the scroll event. `isFocused` is controlled by the onBlur and onFocus event. What do you suggest here?